src/main/java/com/miquido/validoctor/multirule/ConditionalPropertyRule.java

```diff
@@ -27,8 +28,21 @@ public boolean test(PatientType obj) {
   }
 
   @Override
-  public Ailment getAilment() {
-    return rule.getAilment();
+  public Ailment apply(PatientType obj) {
+    if (condition.test(obj)) {
+      return rule.apply(propertyGetter.apply(obj));
+    }
+    return null;
```

Comment:
Returning null is bad practice - high possibility of NullPointerException. I suggest to return optional or throw checked exception when there is no Ailment

Reply:
i do not see why this is an exceptional state? apply's method contract states it should return null if there is no aliment stated. It is called by Validoctor class only so there is no chance of NullPointerException